File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        int[] answer = new int[s.length()];
        
        HashMap<String, Integer> hm = new HashMap<>();
        
        for(int i=0; i<s.length(); i++) {
        	if(hm.keySet().contains(s.charAt(i)+"")) {
        		answer[i] = i - hm.get(s.charAt(i)+"");
        	}else {
        		answer[i] = -1;
        	}
        	hm.put(s.charAt(i)+"", i);
        }
        return answer;
    }
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    answer.append(-1)
    for i in range(1,len(s)):
        cnt = 0
        b_p = 0
        for j in range(i-1,-1,-1):
            cnt += 1
            if s[i] == s[j]:
                answer.append(cnt)
                b_p = 1
                break
        if b_p == 0:
            answer.append(-1)
    return answer